sky/backends/backend_utils.py

```diff
@@ -817,10 +817,20 @@ def write_cluster_config(
     else:
         # ssh_proxy_command_config: Dict[str, str], region_name -> command
         # This type check is done by skypilot_config at config load time.
-        assert region_name in ssh_proxy_command_config, (
-            'Optimizer should only yield regions that are in '
-            'the ssh_proxy_command')
-        ssh_proxy_command = ssh_proxy_command_config[region_name]
+
+        # There are two cases:
+        # 1. region_name may not appear in the config, when we are
+        # provisioning an existing clusters launched on a region and the
+        # region is now excluded from the config; we use None as
+        # ssh_proxy_command, which will be restored to the original value
+        # later by _replace_yaml_dicts().
+        # 2. New clusters will be launched on regions only appears in the
+        # ssh_proxy_command_config guaranteed by
+        # Resources.get_valid_regions_for_launchable();
+        assert (keep_launch_fields_in_existing_config or region_name
+                in ssh_proxy_command_config), (region_name,
+                                               ssh_proxy_command_config)
+        ssh_proxy_command = ssh_proxy_command_config.get(region_name, None)
```

Comment:
```suggestion
    # There are two cases:
    if keep_launch_fields_in_existing_config:
        # (1) We're re-provisioning an existing cluster.
        #
        # We use None for ssh_proxy_command, which will be restored to the
        # cluster's original value later by _replace_yaml_dicts().
        ssh_proxy_command = None
    else:
        # (2) We're launching a new cluster.
        #
        # Resources.get_valid_regions_for_launchable() respects the keys (regions)
        # in ssh_proxy_command in skypilot_config. So here we add an assert.
        assert region_name in ssh_proxy_command_config, (region_name, ssh_proxy_command_config)
        ssh_proxy_command = ssh_proxy_command_config[region_name]
```
This seems clearer, e.g., we don't need to assert in case 1?

Reply:
Done. Thanks!